controllers/factory/k8stools/version.go

```diff
@@ -79,12 +68,9 @@ func IsFSGroupChangePolicySupported() bool {
 	return false
 }
 
-// NewHPAEmptyObject returns HorizontalPodAutoscaler object for given kubernetes version
+// NewHPAEmptyObject returns HorizontalPodAutoscaler object
 func NewHPAEmptyObject(opts ...func(obj client.Object)) client.Object {
```

Comment:
I think, we should remove `client.Object from this function and return exact type. It can be done at follow-up commit.